packages/tables/src/Table/Concerns/CanGroupRecords.php

```diff
@@ -72,6 +72,10 @@ public function groupingDirectionSettingHidden(bool | Closure $condition = true)
 
     public function defaultGroup(string | Group | null $group): static
     {
+        if ($group instanceof Group) {
+            $group->table($this);
+        }
```

Comment:
Can we move this to the place where the defaultGroup gets read so if we introduce closure support this won't break